releaseNotes/Microsoft.Featuremanagement.md

```diff
@@ -4,6 +4,37 @@
 # Microsoft.FeatureManagement.AspNetCore
 [Source code ][source_code_web] | [Package (NuGet)][package_web] | [Samples][samples_web] | [Product documentation][docs]
 
+## 4.0.0-preview2 - March 7, 2024
```

Comment:
Looks like we are listing out every feature/change separately. Is there a reason we don't follow the release notes guidelines?

Reply:
Which guideline do you feel isn't followed? This is primarily built off previous releases in this library. 

Releases for FM have been more than just release notes, but have also acted as "marketing" for the new features. We usually list the major features, then list enhancements/additional changes, then breaking changes. 

In this PR specifically, comments caused me to remove all but one of the "enchancements", so I removed the header. https://github.com/Azure/AppConfiguration/pull/886/commits/e01c3f4bb2389b600e9dc7ef380cb9fc3ff2bffd#diff-062e587e3d06ace0b7748874ffc0d6022532d33be3d8199526fa7aaadd07030dR82

Reply:
Shared the release notes guideline offline. Not a problem at all if we want to change it, but we should discuss it.

I saw the preview1 release notes didn't follow the guideline either. That's exactly the problem. Once we had something like that, people will follow it as precedence. This is exactly the problem the guideline tries to solve.